provider/pkg/autoscaling/autoscaling.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaling
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+// Customize resources in the autoscaling module.
+func ReconfigureResources(p *schema.Provider) {
+	autoscalingGroupReconfigure(mustGetResource(p, "aws_autoscaling_group"))
+}
+
+func autoscalingGroupReconfigure(r *schema.Resource) {
```

Comment:
This approach does *not* generate the default values into the Pulumi program, which is nice, while solving the problem. I was thinking of incubating it here but then considering generalizing. Instead of specifying the fields we can run automatic schema-based detection for fields that need it.